src/ProjectWindow.cpp

```diff
@@ -1882,9 +1893,25 @@ void ProjectWindow::DoZoomFit()
 
 void ProjectWindow::ShowEffectsPanel(Track* track)
 {
+   mEffectsTrackSelectionChanged.Reset();
+
    if(track == nullptr)
       return;
 
+   wxWindowUpdateLocker freeze(this);
+
+   mEffectsTrackSelectionChanged = 
+      track->GetOwner()->Subscribe([this](const TrackListEvent &event)
+      {
+         if(event.mType == TrackListEvent::SELECTION_CHANGE)
+         {
+            auto track = event.mpTrack.lock();
+            mEffectsWindow->SetTrack(track && track->IsSelected() ? track : nullptr);
+         }
```

Comment:
I see what this subscription is doing and I understand why you need something like it.

I question whether this is exactly what we will want.  First, sometimes multiple tracks are selected, as when you select all.  This might, wastefully, visit this function once for each track, rebuilding the window many times, and finally leaving it as for the last selected track.

Would it be better to scan the track list for a unique selected track -- and empty the effect list if there is no such track?

But then -- as I pointed out long ago, critiquing the design proposals -- there is a distinction between selection, which is non-exclusive, and focus (indicated by the yellow border), which is.

I think it makes more sense to show the effect stack for the focused track, which means you should listen for what is now the `EVT_TRACK_FOCUS_CHANGE` event (but, like theme change, ANOTHER thing I want to rewrite with Observer, and I also have a waiting commit for that, which I can cherry-pick and push onto your branch).

@DavidBailes will surely want to test this new feature and be satisfied with keystroke navigation, and again, the notion of a "focused" object, rather than a "selected" one, is in general relevant to the targeting of commands given by keystroke.  See for instance `OnTrackMenu` in `TrackMenus.cpp` which handles Shift+M in default key bindings.  That keyboard shortcut opens the track drop-down menu for the (unique) *focused*, not selected, track.

Reply:
> @DavidBailes will surely want to test this new feature and be satisfied with keystroke navigation

Is there a current design specification for the effects panel which includes the keyboard navigation? I have asked this question on https://github.com/audacity/audacity/issues/2845, but have not yet received a reply.

Concerning whether the effects panel show the effects stack for the focused or selected track. The normal pattern is for a side panel to show some property of a selected item in a list. However, I don't think it fits the current case. A user might want to look at the effect stack without changing which track or track was selected, and each audio track having its own effect button implies this. Note however, that when a user presses an effect button, then I presume the intention is that the focus is automatically moved to the effects panel. This means that the track is not the focus and won't have a yellow outline. Are the designers happy with that?

Concerning the list of effects in the Effects panel. When the figma designs of this were originally shared by Jouni, the keyboard navigation of the list of effects, including the controls in each item, was specified as just a series of tabs. I commented on discord at the time that it would be much better if the keyboard navigation was appropriate for a list. Unfortunately, nobody replied to that comment. It is still my view that the list of effects should have keyboard navigation which is appropriate for a list where each list item contains a number of controls, and in addition, the list is read as being a list by users of screen readers. Given that these requirements have implications for how the code for the list is structured, first writing it so that works only with mouse interaction, and then ensuring that it works properly with keyboard interaction and screen readers will end up involving a lot more work than if it was written with all types of interaction from the start.

Reply:
Also possible:  the sidebar changes only if you press Effects on another track.  (Or delete that track)

The Effects button of that track only is painted as down.